File: plugin/src/main/java/com/github/topi314/lavalyrics/plugin/LyricsConfiguration.java
package com.github.topi314.lavalyrics.plugin;

import com.github.topi314.lavalyrics.LyricsManager;
import com.github.topi314.lavalyrics.api.LyricsManagerConfiguration;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

import java.util.Collection;

@Configuration
public class LyricsConfiguration {

	@Bean
	public LyricsManager audioLyricsManagerSupplier(Collection<LyricsManagerConfiguration> lyricsManagerConfigurations) {
		var manager = new LyricsManager();

		for (var config : lyricsManagerConfigurations) {
			manager = config.configure(manager);
		}

		return manager;
	}

}

File: plugin/src/main/java/com/github/topi314/lavalyrics/plugin/LyricsConfiguration.kt
package com.github.topi314.lavalyrics.plugin

import com.github.topi314.lavalyrics.LyricsManager
import com.github.topi314.lavalyrics.api.LyricsManagerConfiguration
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
open class LyricsConfiguration(
    val config: LavaLyricsConfig
) {

    companion object {
        private val log: Logger = LoggerFactory.getLogger(LyricsConfiguration::class.java)
    }

    @Bean
    open fun audioLyricsManagerSupplier(lyricsManagerConfigurations: Collection<LyricsManagerConfiguration>): LyricsManager {
        log.info("Creating LyricsManager")
        var manager = LyricsManager()

        for (config in lyricsManagerConfigurations) {
            manager = config.configure(manager)
        }

        log.info("Sorting LyricsManagers")
        manager.sortLyricsManagers(config.sources)
        log.info("LyricsManager sorted")
        return manager
    }
}
